tests/unit-core/test-snapshot.c

```diff
@@ -284,11 +284,8 @@ main (void)
     jerry_cleanup ();
 
     test_exec_snapshot (snapshot_buffer, snapshot_size, JERRY_SNAPSHOT_EXEC_ALLOW_STATIC);
-  }
 
-  /* Merge snapshot */
-  if (jerry_feature_enabled (JERRY_FEATURE_SNAPSHOT_SAVE) && jerry_feature_enabled (JERRY_FEATURE_SNAPSHOT_EXEC))
-  {
```

Comment:
I think that instead of merging these two blocks, which are actually two separate test cases that happen to be guarded by the same condition, all test cases of `main` should be organized into separate test functions. 

interestingly, at the end of `main`, there are already calls to three test cases organized into functions (`test_function_snapshot`, `test_function_arguments_snapshot`, `test_snapshot_with_user`). but for some reason, the first three test cases are still big blocks with main (prefixed with comments `/* Static snapshot */`, `/* Merge snapshot */`, and `/* Save literals */`). let's refactor them.